liquidity-bot/src/lib.rs

```diff
@@ -201,8 +221,19 @@ impl LiquidityBotTask {
             tokio::select! {
                 event = self.command_rx.recv() => {
                     match event {
-                        Some(Command::WalletEvent(WalletEvent::BlockProcessed { received_tx_outs, .. })) => {
-                            // TODO look at spent TxOuts and see if they match any SCIs we submitted
+                        Some(Command::WalletEvent(WalletEvent::BlockProcessed { received_tx_outs, spent_tx_outs, .. })) => {
+                            // Look at spent TxOuts and see if they match any SCIs we submitted
+                            match self.look_for_fulfilled_scis(&spent_tx_outs, &received_tx_outs) {
+                                Ok(fulfilled_scis) => {
+                                    for fulfilled_sci in fulfilled_scis {
+                                        metrics::PER_TOKEN_METRICS.update_metrics_from_fulfilled_sci(&fulfilled_sci);
+                                    }
+                                }
+
+                                Err(err) => {
+                                    log::error!(self.logger, "Error looking at spent TxOuts: {}", err);
```

Comment:
Would it be helpful here to return the list of spent listed txouts in the error or split that out into a separate function? I think that first section never errors, and always removes the listed tx outs, which might make going through the listed txouts to match against the spent_tx_outs difficult later.

Reply:
I'm not sure what you are referring to by the first section or returning the spent list at this point in the code. Does this comment need to go somewhere else?

As a side note, `look_for_fulfilled_scis` is only expected to fail if the data passed to it is wrong, not if it cannot make sense of a particular spend. As such I think it's fine if when that happens metrics aren't updated and an error-level log message is outputted. Its actionable and we should look into why that happened.

Reply:
Sorry about being unclear. I was just trying to observe that inside look for fulfilled scis that the first codeblock here: https://github.com/mobilecoinofficial/deqs/blob/3b08b83ed19fcd05f75c509c194540f4a4eee25b/liquidity-bot/src/lib.rs#L306 removes scis from self.listed_scis. And then uses the list of removed scis for various checks. I was wondering if because those scis have been removed already, it might be hard to rebuild the state when looking at the error log later.

Reply:
Thats a bit of an involved change so I will do it in a followup PR.

Reply:
https://github.com/mobilecoinofficial/deqs/pull/43